spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TimeoutInterceptor.java

```diff
@@ -51,41 +51,57 @@ public void run() {
         long waitMillis = timeout.unit().toMillis(timeout.value());
         boolean synced = false;
 
+        boolean syncedWithFeature = false;
         try {
           startLatch.countDown();
-          startLatch.await(5, TimeUnit.SECONDS);
+          syncedWithFeature = startLatch.await(5, TimeUnit.SECONDS);
         } catch (InterruptedException ignored) {
+          // this is our own thread, so we can ignore the interruption safely
+        }
+        if (!syncedWithFeature) {
           System.out.printf("[spock.lang.Timeout] Could not sync with Feature for method '%s'", invocation.getMethod().getName());
         }
 
+        while (waitMillis > 0) {
+          long waitStart = System.nanoTime();
+          try {
+            synced = sync.offer(stackTrace, waitMillis, TimeUnit.MILLISECONDS);
+          } catch (InterruptedException ignored) {
+            // this is our own thread, so we can ignore the interruption safely and continue the remaining waiting
+            waitMillis -= TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - waitStart);
+            continue;
+          }
+          break;
+        }
```

Comment:
If it indeed happens that the watcher thread is interrupted during the initial wait, it should continue the initial wait, shouldn't it?
Or should it really forward the interrupting to the Feature thread, just because it got interrupted for whatever reason?